blocks/faq/faq.js

```diff
@@ -76,24 +77,21 @@ export default async function decorate(block) {
   const faqTitle = createElement('div', 'faq-title');
   const faqContent = createElement('div', 'faq-content');
   const faqButton = createElement('div', 'more-button');
-  let title = '';
-  let buttonTitle = '';
-  let expendButtonTitle = '';
-  [...block.children].forEach((child, i) => {
-    if (i === 0) {
-      title = [...child.children][1].innerHTML;
-    } else if (i === 1) {
-      buttonTitle = [...child.children][1].innerHTML;
-    } else if (i === 2) {
-      expendButtonTitle = [...child.children][1].innerHTML;
-    } else {
-      if ([...child.children].length === 3) {
-        [...child.children][0].remove();
-      }
+  const blockConfig = readBlockConfig(block);
+  const topTitle = blockConfig.title;
+  const buttonTitle = blockConfig['button-title'];
+  const expendButtonTitle = blockConfig['expend-button-title'];
+  let faqContentIndex = false;
+  [...block.children].forEach((child) => {
```

Comment:
you can consider using readBlockMarkup method ?

Reply:
@anagarwa A good point, I saw that method. For FAQ, we have a case to show an icon in the title, it will need some effort to refactoring the logic if we use that method. For now, I just keep it as is.
 
<img width="402" alt="image" src="https://github.com/aemsites/icicidirect/assets/57093988/4f94ec44-897b-4681-908c-1c0285bf0faa">

Reply:
@JiangLong2019 may be open a follow up jira, to track if that is something we can pick up in future ?

Reply:
@anagarwa Sure, I think not only this block, others we can do the same. I've raised a issue for it - https://github.com/aemsites/icicidirect/issues/164